agent/base.py

```
@@ -0,0 +1,106 @@
+            },
+            config={
+                "configurable": {
+                    "recursion_limit": self.recursion_limit,
+                    "block_height": block_height,
+                }
+            },
+            prompt_cache_key=prompt_cache_key
+        )
+        return response
+
+    async def ainvoke(self, input: dict):
```

Ruff (0.15.10)

[error] Function argument `input` is shadowing a Python builtin

(A002)